content/What's good about our iOS-base - German Lopez.md

```diff
@@ -0,0 +1,69 @@
+
+
+## The beginnings
+
+The idea was simple, we wanted an XCode project that allowed us to jump right into feature development, maybe with slight modifications or architecture changes. We started putting together everything we knew we were using in every new app, including a networking layer, a couple of handy dependencies, view customization helpers, and a bunch of useful Swift extension. We even added example features for log-in and sign up users into the app and with local session storage.
+That was it, not very ambitious but it saved a lot of setup time and of course, we now had something to collaborate and improve over time.
+
+### No attachments
+
+The project includes basic modules like session storage, routing, analytics and comes with MVVM architecture out of the box, but there is nothing that forces the developers to use this pattern or any other aspect of the main project.
+As an engineer, you need to chose the right architecture for your application and our iOS-base allows you to do that with no hassle.
+
+### Code quality
+
+One of the implicit goals of this project was to centralize code quality standards for all iOS developers, and directly contribute to our company goals.
+
+As a team, we reached an agreement on the requirements for a good code quality/best practices and made that a pillar in our code base. As a result we decided to:
+
+  -   Adopted [Google’s swift style guides](https://github.com/google/swift).
+  -   Configured [SwiftLint](https://github.com/realm/SwiftLint) to run locally and in the CI platform of preference. SwiftLint rules respond to our agreement and the style guide.
+  -   Integrated CodeClimate to ensure the best coding practices.
+
+
+## Things changed... for good
+
+I would be sceptical if someone said that just by starting a new project from a boilderplate, it improved the productivity by a magical percentage.
+The truth is that it really improved, not just productivity, but other aspects of our engineering team.
+- The intial iterations finished earlier, generating confidence in our clients.
+- The iOS-base plus a good shared-knowlegde base in our company, helped to develop similar features in record time.
+- As an open source project it generated interest in the iOS team members and increased their willingness to collaborate.
+- The project gave more visibility to our company from the tech point of view.
+#
+
+# Take away
+
+Most people will ask, why creating _another_ iOS boilerplate project?
+Yes, there are a few thousand open source projects out there with similar purposes. Some of them with more functionalities, some more specific, you name it.
+The answer to that is that you will never find the one that solves all of your problems.
+
+Of course you can fork the 'best match' and adapt it to your needs, but you will be missing the opportunity to make that **your own** solution.
+
+### Key points if you decide to build your own:
```

Comment:
Maybe add a `keep it lean` point (?) something on the lines of:
```
**Keep it lean** 
Every project has different needs, so try to avoid adding features to your base that ties you to a specific implementation or service.
```

Reply:
Sounds good. BTW you mean `clean` right?

Reply:
No 😬 , I meant lean in the sense of keeping things flexible and dynamic, but clean works as well.

Reply:
Oh I didn't know that meaning for lean. Isn't that 'inclinar' o 'apoyar'?

Reply:
Yep, that as well, but lean is used as: `Leanness means eliminating all waste, including time, and ensuring a smooth and predictable output.`  So something that's easy to work with and doesn't have too much overhead